### PR TITLE
AA-492: Add event data for consumption in the backend

### PR DESCRIPTION
**Pull request by Dillon-Dumesnil**
Related to https://github.com/edx/edx-platform/pull/26264

Important, needs to be merged AFTER the above otherwise the endpoint will start erroring due to the new parameter.

### CODE REVIEW

File: src/course-home/data/thunks.js [hunk 1]
```diff
@@ -111,9 +111,10 @@ export async function saveCourseGoal(courseId, goalKey) {
 
 export function processEvent(eventData, getTabData) {
   return async (dispatch) => {
+    const { research_event_data: researchEventData } = eventData;
```
Comment (Dillon-Dumesnil) [inline src/course-home/data/thunks.js:116]: Pulling this out early so the data doesn't get camelCased and is easier to use when it's passed to the backend.

Comment (mikix) [reply src/course-home/data/thunks.js:116]: Maybe that comment is worth being in-code? But I don't feel strongly - a dev would probably see that.

**Review by mikix — APPROVED** [review]

File: src/course-home/data/thunks.js (hunk 1 above)
Comment (Dillon-Dumesnil) [reply src/course-home/data/thunks.js:116]: Yeah, that's what I'm hoping.

Comment (Dillon-Dumesnil) [reply src/course-home/data/thunks.js:116]: I'll just update it. Need to change edx-platform anyway

**Pull request merged**